solr/core/src/java/org/apache/solr/response/transform/ChildDocTransformerFactory.java

```diff
@@ -61,6 +65,10 @@
  */
 public class ChildDocTransformerFactory extends TransformerFactory {
 
+  public static final String PATH_SEP_CHAR = "/";
+  public static final String NUM_SEP_CHAR = "#";
+  private static final String sRootFilter = "*:* NOT " + NEST_PATH_FIELD_NAME + ":*";
```

Comment:
Oh I forgot.  Perhaps instead create the Lucene Query directly (rather than create a String that needs to be parsed). Use a BooleanQuery with a MUST_NOT clause around DocValuesFieldsExistsQuery for the nest path field?   And comment this only applies to schemas that are enabled for this enhanced nesting.